### DISCUSSION TITLE
Retaining a portion of the original filename in the output filename

### DISCUSSION THREAD

**Discussion by lynxmeninx** [ImageMagick/ImageMagick#5337] [Help] [answered]
Hi there! I'm really sorry to ask a potentially redundant question - I'm certain this has been asked before - but I'm having a **lot** of trouble wording my issue so i'd really appreciate being pointed in the right direction either way ^_^/ I'm using IM 7.1.0 43 Q16(?) on Windows x64.

I have a ton of gifs of characters that i'm converting into spritesheets. All of the gifs are named things like "stand_[charactername]" and "walk_[charactername]". Here's my code that works perfectly thus far: 

```
magick convert ^
   ( stand_*.gif -coalesce +append ) ^
   ( walk_*.gif -coalesce +append ) ^
   -append -filter point -resize 50% %d_spritesheet.png 
```

I plan on putting the output all in the same folder (although I'm also having trouble using -path, but I feel more confident in troubleshoothing that), but I want the output filenames to be more like [charactername]_spritesheet.png instead of [number]_spritesheet.png.
Can this be done? What can I replace "%d" with to make this happen?

You might be able to tell I don't know the first thing about coding or programming -- thank you for your patience!

**Comment by fmw42** [comment]
You cannot have wild card names for both the input and output. So you may need to write a script "FOR" loop over each character and append the stand and walk into new files. Then append all the files or use montage to combined them. Also in Imagemagick 7, use magick, not magick convert.  Sorry, I do not program Windows, so my Unix scripts will do you no good.

**Reply by lynxmeninx** [reply]
Thank you so much!

**Comment by snibgo** [comment, marked as answer]
I suggest using just "magick" instead of "magick convert".

> I plan on putting the output all in the same folder (although I'm also having trouble using -path, but I feel more confident in troubleshoothing that), but I want the output filenames to be more like [charactername]_spritesheet.png instead of [number]_spritesheet.png.
Can this be done? What can I replace "%d" with to make this happen?

Don't use `-path`. Just put the full path in the output filename, eg "`myfolder\%d_spritesheet.png`"

On the character names (ie some part of the input filenames) in the output filenames, no, IM can't do that directly. You might use shell facilities to make a unique list of character names, then loop through those in a script.

**Reply by lynxmeninx** [reply]
Thank you so much! I wonder where I got the idea to use magick convert instead of just magick. Could that have been causing problems?

**Comment by snibgo** [comment]
"`magick convert`" applies the old v6 syntax, instead of current v7 syntax. This wouldn't cause problems in your command. v7 is better. v6 is provided only for backwards compatibility.
